frontend/src/stores/knowledges/actions.ts

```diff
@@ -0,0 +1,47 @@
+import { get, post } from '@/services/http'
+import { getKnowledge } from './getters'
+import { knowledges } from './state'
+import type { Knowledge } from '@/domain/Knowledge'
+
+async function fetchKnowledges(): Promise<void> {
+  if (getKnowledge?.value?.length > 0) {
+    return
+  }
+
+  const knowledgeArray = await get<Knowledge[]>('/knowledge')
+  setKnowledges(knowledgeArray)
+}
+
+async function addKnowledge(knowledge: Knowledge): Promise<void> {
+  await post<Knowledge[]>('/knowledge', knowledge)
+  knowledges.value.push(knowledge)
+}
+
+function setKnowledges(newKnowledges: Knowledge[]): void {
+  knowledges.value = newKnowledges
+}
+
+export const saveKnowledge = async (
+  name: string,
+  roleId: number,
+  levelId: number,
+  weight: number
+): Promise<void> => {
+  try {
+    const knowledge: Knowledge = {
+      name,
+      roleId: roleId.toString(),
+      levelId: levelId.toString(),
+      weight: weight.toString()
+    }
+
+    await addKnowledge(knowledge)
```

Comment:
Why not to do everything inside the same function? addKnowledge and saveKnowledge appears to be the same thing to me, no?